### ISSUE TITLE
a issue in case of sequential input messages with no intervals.

### ISSUE THREAD

**Issue by stanakaj** [Daniel-t/node-red-contrib-aws#60]
Thank you very much for your great work.
I found a problem when it goes with sequential messages with no intervals.

# Symptom summary
The msg object from S3 Node has unexpected fields, when you use sequential input messages with no intervals.

# Details with flow sample
![sample-flow](https://user-images.githubusercontent.com/44893868/65577755-6c414b00-dfaf-11e9-826a-817941039200.PNG)

## a single message : OK
the upper line in the sample flow is for a single message.
- start with "single" inject node.
- Input @ "single input"
> {"_msgid":"8a60a4c9.9dd2f8","topic":"","payload":1569393751386,"Key":"a.csv"}
- Output @ "output"
msg.payload : expected
msg._msgid : expected

> {
> 	"_msgid": "b8886cb9.bcd4b",
> 	"topic": "",
> 	"payload": { ... },
> 	"Key": "a.csv"
> }

## mulitple messages : NG
the lower line in the sample flow is for sequential messages with no intervals.
- start with "multi" inject node.
- Input A @ "multi input"
> {"topic":"","payload":{"Key":"a.csv"},
> "parts":{"id":"b614d016.2980f","type":"array","count":3,"len":1,"index":0},
> "_msgid":"dfaf7c7c.b5324","Key":"a.csv"}
- input B @ "multi input"
> {"topic":"","payload":{"Key":"b.csv"},
> "parts":{"id":"b614d016.2980f","type":"array","count":3,"len":1,"index":1},
> "_msgid":"bc24a2ec.e3f25","Key":"b.csv"}
- input C @ "multi input"
> {"topic":"","payload":{"Key":"c.csv"},
> "parts":{"id":"b614d016.2980f","type":"array","count":3,"len":1,"index":2},
> "_msgid":"1f1f1b8f.093034","Key":"c.csv"}

- OutputA @ "output" : NG
msg.payload : expected
**msg._msgid : not expected
msg.Key : not expected**

> {
> 	"topic": "",
> 	"payload": {
> 		"AcceptRanges": "bytes",
> 		"LastModified": "2019-08-01T17:35:29.000Z",
> 		"ContentLength": 173,
> 		"ETag": "\"1480d7f550186bd442d223ac1500039e\"",
> 		"ContentType": "",
> 		"Metadata": {},
> 		"Body": [...]
> 	},
> 	"parts": {...},
> 	"_msgid": "1f1f1b8f.093034",
> 	"Key": "c.csv"
> }

- OutputB @ "output" : NG
msg.payload : expected
**msg._msgid : not expected
msg.Key : not expected**

> {
> 	"topic": "",
> 	"payload": {
> 		"AcceptRanges": "bytes",
> 		"LastModified": "2019-08-01T17:35:30.000Z",
> 		"ContentLength": 149,
> 		"ETag": "\"b1bf93e42826eb1d94f4596dc9b37993\"",
> 		"ContentType": "",
> 		"Metadata": {},
> 		"Body": [...]}
> 	},
> 	"parts": {...},
> 	"_msgid": "1f1f1b8f.093034",
> 	"Key": "c.csv"
> }

- OutputC @ "output" : OK
msg.payload : expected
msg._msgid : expected
msg.Key : expected

> {
> 	"topic": "",
> 	"payload": {
> 		"AcceptRanges": "bytes",
> 		"LastModified": "2019-08-01T17:35:30.000Z",
> 		"ContentLength": 294,
> 		"ETag": "\"c109738b12d564c47d1637a6c4ac806f\"",
> 		"ContentType": "",
> 		"Metadata": {},
> 		"Body": [...]
> 	},
> 	"parts": {...},
> 	"_msgid": "1f1f1b8f.093034",
> 	"Key": "c.csv"
> }

## The causes in code, I guess.
for example, in the S3 node, below code. I see the "node.sendMsg" uses a last bound msg object at any time.
```
		node.on("input", function(msg) {
			node.sendMsg = function (err, data) {
				if (err) {
				    node.status({fill:"red",shape:"ring",text:"error"});
				   node.error("failed: " + err.toString(), msg);
				   node.send([null, { err: err }]);
    				return;
				} else {
				msg.payload = data;
				node.status({});
				}
				node.send([msg,null]);
			};
```

## Sample Flow 
[sample-flow.json.zip](https://github.com/Daniel-t/node-red-contrib-aws/files/3651231/sample-flow.json.zip)

**Comment by soares-luan** [comment]
Same problem, with a loop, my last message is reused on the new call of s3 getHead

**Comment by Daniel-t** [comment]
Related to #66 and a fix proposed in #67, however needs to be expanded to all nodes.

**Comment by Daniel-t** [comment]
Should be fixed
